src/wallet/rpcdump.cpp

```diff
@@ -323,6 +323,103 @@ RPCHelpMan importaddress()
+    if (!isRedeemScript && ismine == ISMINE_SPENDABLE) {
+        throw JSONRPCError(RPC_WALLET_ERROR, "The wallet contains the private key for this address");
+    } else if (!isRedeemScript && ismine == ISMINE_NO) {
+        throw JSONRPCError(RPC_WALLET_ERROR, "The wallet does not contain this address or script");
+    }
+    CHECK_NONFATAL(ismine == ISMINE_WATCH_ONLY || isRedeemScript);
+
+    LegacyScriptPubKeyMan& spk_man = EnsureLegacyScriptPubKeyMan(*pwallet);
+
+    if (spk_man.HaveWatchOnly(script) && !spk_man.PurgeWatchOnly(script)) {
+        throw JSONRPCError(RPC_WALLET_ERROR, "Error removing address/script from wallet");
+    }
+    if (isRedeemScript) {
+        CScriptID scriptID = CScriptID(script);
+        if (!spk_man.HaveCScript(scriptID)) {
+            throw JSONRPCError(RPC_WALLET_ERROR, "Error removing p2sh redeemScript from wallet");
+        }
+        RemoveAddress(pwallet, ScriptHash(scriptID), purgeTxs);
+    } else {
+        CTxDestination destination;
+        if (ExtractDestination(script, destination)) {
+            pwallet->DelAddressBook(destination);
+        }
+    }
+
+    if (purgeTxs) {
+        pwallet->RemoveTransactions(script);
+    }
+}
+
+static void RemoveAddress(CWallet* const pwallet, const CTxDestination& dest, bool purgeTxs) EXCLUSIVE_LOCKS_REQUIRED(pwallet->cs_wallet)
+{
+    CScript script = GetScriptForDestination(dest);
+    RemoveScript(pwallet, script, false, purgeTxs);
+    // remove from address book
+    if (IsValidDestination(dest)) {
+        pwallet->DelAddressBook(dest);
+    }
+}
+
+RPCHelpMan removeaddress()
+{
+    return RPCHelpMan{
+        "removeaddress",
+        "\nRemoves an address or script (in hex) that was being watched as if it were in your wallet but was not being used to spend. Requires a new wallet backup.\n",
+        {
+            {"address", RPCArg::Type::STR, RPCArg::Optional::NO, "The Bitcoin address (or hex-encoded script)"},
+            {"p2sh", RPCArg::Type::BOOL, /* default */ "false", "Remove the P2SH version of the script as well"},
+            {"purgetransactions", RPCArg::Type::BOOL, /* default_val */ "false", "Remove existing transactions from the wallet"},
+        },
+        RPCResult{RPCResult::Type::NONE, "", ""},
+        RPCExamples{
+            "\nRemove an address\n" + HelpExampleCli("removeaddress", "\"myaddress\"") +
+            "\nAs a JSON-RPC call\n" + HelpExampleRpc("removeaddress", "\"myaddress\"")},
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue {
+            std::shared_ptr<CWallet> const wallet = GetWalletForJSONRPCRequest(request);
+            if (!wallet) return NullUniValue;
+            CWallet& pwallet = *wallet.get();
+
+            EnsureLegacyScriptPubKeyMan(pwallet, true);
+
+            // Whether to remove the p2sh version too
+            bool fP2SH = false;
+            if (!request.params[1].isNull()) {
+                fP2SH = request.params[1].get_bool();
+            }
```

Comment:
can replace these lines (and have constness and follow variable naming style per doc/developer-notes.md) with
```c++
const bool is_p2sh{request.params[1].isNull() ? false : request.params[1].get_bool()};
```